Add hostname detection for azure cloud provider

hush-hush opened the pull request:
### What does this PR do?

- Add detection for azure VM ID using the metadata service
- fix hostname in metadata payload
- fix `resources` serialization who was shadowing the `meta` section from the host metadata
- small typo and logging improvements

hush-hush committed:
Adding Azure VM ID to the 'host_aliases' metadata payload

masci reviewed (approved):
Left a small comment but feel free to merge, LGTM

hush-hush committed:
Fix cache key for hostname

The hostname is set with a prefix in the agent cache. Forgetting the prefix result in an empty hostname in metadata.

hush-hush committed:
Fix typo in metadata struct

hush-hush committed:
Fix 'resources' metadata serialization

The resources payload should be in a 'resources' section in the json to follow agent5 format.
This also causes the json to have 2 'meta' sections that would overwrite each other resulting
in the 'meta' section from host payload to disappear in the final JSON payload.

masci reviewed (approved):
LGTM, pinging @olivielpeau since the fix to the metadata payload might work for him too

olivielpeau reviewed (approved):
One nitpicky comment, but LGTM overall, feel free to merge! 👍 

And thanks for making these fixes, I'll give them a go once this is merged
